app/controllers/invitations/CheckYourAnswersController.scala

```diff
@@ -83,21 +84,14 @@ class CheckYourAnswersController @Inject()(appConfig: FrontendAppConfig,
             inviteeName,
             getExpireAt
           )
-          isSchemeAssociatedWithInvitee(request.psaId.id, schemeDetails.srn, inviteePsaId).flatMap {
-            case true =>
-              Future.successful(Redirect(routes.PsaAlreadyAssociatedController.onPageLoad()))
-            case _ =>
-              invitationConnector.invite(invitation)
-                .map(_ => Redirect(navigator.nextPage(CheckYourAnswersId(schemeDetails.srn), NormalMode, request.userAnswers)))
-                .recoverWith {
-                  case _: NotFoundException =>
-                    Future.successful(Redirect(controllers.invitations.routes.IncorrectPsaDetailsController.onPageLoad()))
-                }
+
+          isSchemeAssociatedWithInvitee(request.psaId.id, schemeDetails.srn, inviteePsaId).flatMap { isAssociated =>
+            invite(invitation, schemeDetails).map(result =>
+              if (isAssociated) Redirect(routes.PsaAlreadyAssociatedController.onPageLoad()) else result
+            )
           }.recoverWith {
             case _: NameMatchingFailedException =>
               Future.successful(Redirect(controllers.invitations.routes.IncorrectPsaDetailsController.onPageLoad()))
-            case _: PsaAlreadyInvitedException =>
-              Future.successful(Redirect(controllers.invitations.routes.InvitationDuplicateController.onPageLoad()))
             case _ =>
```

Comment:
Leaving this case means that we swallow errors and don't recover the user. We should just remove it and let the error bubble up so that we can see that something has gone wrong.

Reply:
done